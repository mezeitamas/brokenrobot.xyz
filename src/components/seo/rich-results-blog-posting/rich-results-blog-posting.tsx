import React from 'react';
import type { FunctionComponent } from 'react';

import type { Person, Organization, WebSite, BlogPosting, WithContext } from 'schema-dts';

import type { Author } from '../../site-metadata/site-metadata.types';

type RichResultsBlogPostingProps = {
    siteName: string;
    siteUrl: string;
    title: string;
    description: string;
    url: string;
    published: string;
    author: Author;
};

const RichResultsBlogPosting: FunctionComponent<RichResultsBlogPostingProps> = ({
    siteName,
    siteUrl,
    title,
    description,
    url,
    published,
    author
}) => {
    const authorPerson: Person = {
        '@type': 'Person',
        '@id': `${siteUrl}/about-me/#Person`,
        name: author.name,
        url: `${siteUrl}/about-me/`,
        sameAs: [
            author.social.githubUrl,
            author.social.linkedinUrl,
            author.social.mastodonUrl,
            author.social.twitterUrl
        ]
    };

    const publisher: Organization = {
        '@type': 'Organization',
        '@id': `${siteUrl}/#Organization`,
        name: siteName,
        url: siteUrl
    };

    const website: WebSite = {
        '@type': 'WebSite',
        '@id': `${siteUrl}/#WebSite`,
        name: siteName,
        url: siteUrl,
        inLanguage: 'en-US',
        author: [authorPerson],
        publisher: publisher
    };

    const blogPosting: WithContext<BlogPosting> = {
        '@context': 'https://schema.org',
        '@type': 'BlogPosting',
        '@id': `${url}#BlogPosting`,
        headline: title,
        name: title,
        description: description,
        datePublished: published,
        dateModified: published,
        url: url,
        inLanguage: 'en-US',
        author: [authorPerson],
        publisher: publisher,
        isPartOf: website
    };

    return <script type="application/ld+json">{JSON.stringify(blogPosting)}</script>;
};

export { RichResultsBlogPosting };

export type { RichResultsBlogPostingProps };
