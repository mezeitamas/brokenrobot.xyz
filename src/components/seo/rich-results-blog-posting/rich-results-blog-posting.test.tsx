import React from 'react';

import { create } from 'react-test-renderer';

import { RichResultsBlogPosting } from './rich-results-blog-posting';
import type { RichResultsBlogPostingProps } from './rich-results-blog-posting';

describe('<RichResultsBlogPosting />', () => {
    test('should render', () => {
        // Given
        const props: RichResultsBlogPostingProps = {
            siteName: 'Broken Robot',
            siteUrl: 'https://www.brokenrobot.xyz',
            title: 'Hosting a static website on Amazon S3',
            description:
                'Hosting a simple static website on Amazon Web Services (AWS) can be daunting at first, but luckily itâ€™s quite straightforward. One option...',
            url: 'https://www.brokenrobot.xyz/blog/hosting-a-static-website-on-amazon-s3/',
            published: '2023-05-01T10:10:05.000Z',
            author: {
                name: 'Tamas Mezei',
                social: {
                    githubUrl: 'https://github.com/mezeitamas',
                    linkedinUrl: 'https://www.linkedin.com/in/mezeitamas/',
                    twitterUrl: 'https://twitter.com/brokenrobot_xyz',
                    twitterUsername: '@brokenrobot_xyz'
                }
            }
        };

        // When
        const component = create(<RichResultsBlogPosting {...props} />).toJSON();

        // Then
        expect(component).toMatchSnapshot();
    });
});
