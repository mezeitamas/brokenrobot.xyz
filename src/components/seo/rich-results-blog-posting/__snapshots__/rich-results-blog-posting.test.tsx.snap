// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RichResultsBlogPosting /> should render 1`] = `
<script
  type="application/ld+json"
>
  {"@context":"https://schema.org","@type":"BlogPosting","@id":"https://www.brokenrobot.xyz/blog/hosting-a-static-website-on-amazon-s3/#BlogPosting","headline":"Hosting a static website on Amazon S3","name":"Hosting a static website on Amazon S3","description":"Hosting a simple static website on Amazon Web Services (AWS) can be daunting at first, but luckily itâ€™s quite straightforward. One option...","datePublished":"2023-05-01T10:10:05.000Z","dateModified":"2023-05-01T10:10:05.000Z","url":"https://www.brokenrobot.xyz/blog/hosting-a-static-website-on-amazon-s3/","inLanguage":"en-US","author":[{"@type":"Person","@id":"https://www.brokenrobot.xyz/about-me/#Person","name":"Tamas Mezei","url":"https://www.brokenrobot.xyz/about-me/","sameAs":["https://github.com/mezeitamas","https://www.linkedin.com/in/mezeitamas/"]}],"publisher":{"@type":"Organization","@id":"https://www.brokenrobot.xyz/#Organization","name":"Broken Robot","url":"https://www.brokenrobot.xyz"},"isPartOf":{"@type":"WebSite","@id":"https://www.brokenrobot.xyz/#WebSite","name":"Broken Robot","url":"https://www.brokenrobot.xyz","inLanguage":"en-US","author":[{"@type":"Person","@id":"https://www.brokenrobot.xyz/about-me/#Person","name":"Tamas Mezei","url":"https://www.brokenrobot.xyz/about-me/","sameAs":["https://github.com/mezeitamas","https://www.linkedin.com/in/mezeitamas/"]}],"publisher":{"@type":"Organization","@id":"https://www.brokenrobot.xyz/#Organization","name":"Broken Robot","url":"https://www.brokenrobot.xyz"}}}
</script>
`;
